mithril-aggregator/src/services/prover.rs

```diff
@@ -245,47 +468,136 @@ mod tests {
+                retriever_mock
+                    .expect_get_by_hashes()
+                    .with(eq(transaction_hashes_to_prove))
+                    .return_once(move |_| Ok(transactions_to_prove));
+
+                let block_ranges_to_prove = test_data.block_ranges_to_prove.clone();
+                let all_transactions_in_block_ranges_to_prove =
+                    test_data.all_transactions_in_block_ranges_to_prove.clone();
+                retriever_mock
+                    .expect_get_by_block_ranges()
+                    .with(eq(block_ranges_to_prove))
+                    .return_once(move |_| Ok(all_transactions_in_block_ranges_to_prove));
+            },
+            |block_range_root_retriever_mock| {
+                let block_ranges_map = test_data.block_ranges_map.clone();
+                block_range_root_retriever_mock
+                    .expect_compute_merkle_map_from_block_range_roots()
+                    .return_once(|_| {
+                        Ok(test_data::compute_mk_map_from_block_ranges_map(
+                            block_ranges_map,
+                        ))
+                    });
+            },
+        );
 
         let transactions_set_proof = prover
-            .compute_transactions_proofs(&fake_data::beacon(), &transaction_hashes)
+            .compute_transactions_proofs(&test_data.beacon, &test_data.transaction_hashes_to_prove)
             .await
             .unwrap();
 
         assert_eq!(transactions_set_proof.len(), 1);
         assert_eq!(
             transactions_set_proof[0].transactions_hashes(),
-            &transaction_hashes[0..=2].to_vec()
+            transaction_hashes_known
         );
         transactions_set_proof[0].verify().unwrap();
     }
 
     #[tokio::test]
-    async fn cant_compute_proof_if_retriever_fail() {
-        let (transaction_hashes, _transactions) = generate_transactions(3);
-        let prover = build_prover(|retriever_mock| {
-            retriever_mock
-                .expect_get_by_hashes()
-                .with(eq(transaction_hashes.clone()))
-                .returning(|_| Err(anyhow!("Error")));
-        });
+    async fn cant_compute_proof_if_transaction_retriever_fails() {
+        let total_block_ranges = 5;
+        let total_transactions_per_block_range = 3;
+        let transactions = test_data::generate_transactions(
+            total_block_ranges,
+            total_transactions_per_block_range,
+        );
+        let transactions_to_prove =
+            test_data::filter_transactions_for_indices(&[1, 2, 4], &transactions);
+        let test_data = test_data::build_test_data(&transactions_to_prove, &transactions);
+        let prover = build_prover(
+            |retriever_mock| {
+                let transaction_hashes_to_prove = test_data.transaction_hashes_to_prove.clone();
+                retriever_mock
+                    .expect_get_by_hashes()
+                    .with(eq(transaction_hashes_to_prove))
```

Comment:
I don't think that you need to check that here. The other tests already do that for us so you can remove this `with` condition (same comment for the other 'fail' test below).

Reply:
Done